File: api/types/job.ts
export enum JobStatus {
    Pending = 'pending',
    InProgress = 'in-progress',
    Completed = 'completed',
    Failed = 'failed',
  }
export enum TaskType {
    Fetch = 'fetch',
}
export interface Job {
    id: string;
    task: TaskType;
    config: Record<string, any>;
    
}

export interface JobResponse {
    key: string;
    status: JobStatus;
}

File: simple-go-worker/core/job.go
package core

// Define constants for job statuses
const (
	Pending    = "pending"
	InProgress = "in-progress"
	Completed  = "completed"
	Failed     = "failed"
)

type Job struct {
	ID     string                 `json:"id""`
	Task   string                 `json:"task"`
	Config map[string]interface{} `json:"config"`
}

type JobResponse struct {
	Key    string `json:"key"`
	Status string `json:"status"`
}
